extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/HttpSecurityRecorder.java

```diff
@@ -362,4 +373,47 @@ private synchronized void setPatchMatchingPolicyEnabled() {
 
         protected abstract boolean httpPermissionsEmpty();
     }
+
+    public void setMtlsCertificateRoleProperties(HttpConfiguration config) {
+        if (config.auth.certificateRoleProperties.isPresent()) {
+            Map<String, Set<String>> roles = new HashMap<>();
+
+            Path rolesPath = config.auth.certificateRoleProperties.get();
+            URL rolesResource = null;
+            if (Files.exists(rolesPath)) {
+                try {
+                    rolesResource = rolesPath.toUri().toURL();
+                } catch (MalformedURLException e) {
+                    throw new ConfigurationException(
+                            "quarkus.http.auth.mtls-certificate-role-properties is malformed",
```

Comment:
One thing I'm not sure about is whether it should be `quarkus.http.auth.certificate-role-properties` or `quarkus.http.ssl.certificate-role-properties` - I did not put it into the `.ssl.` one because I thought it was about  (M)TLS specifics. 
The other thing is that I should probably rename it to `certificate-role-mappings`